packages/botonic-plugin-text-classification/src/process/text-processor.ts

```diff
@@ -1,5 +1,11 @@
+import { LabelEncoder } from '@botonic/nlp/lib/encode/label-encoder'
+import {
+  PADDING_TOKEN,
+  UNKNOWN_TOKEN,
+} from '@botonic/nlp/lib/preprocess/constants'
 import { Preprocessor } from '@botonic/nlp/lib/preprocess/preprocessor'
 import type { Tensor2D } from '@tensorflow/tfjs'
+import { tensor } from '@tensorflow/tfjs'
 
 export class TextProcessor {
```

Comment:
this class is very similar to ner's TextProcessor. does it make sense to move it to botonic-nlp and adapt it to each task with a decorator or strategy design pattern?

Reply:
Both tasks of the nlp package contain a `Processor` class that is responsible for creating the Input and expected Output tensors.

Also, in the same way, each plugin of both NLP tasks (text-classification and ner) contains a `TextProcessor` class which is responsible for creating **only** the Input tensor.

One important fact is that the creation process of the Input tensor is the same for all implemented NLP tasks for the moment, but the expected Output creation is different for each one.

I think if maybe we should split `Processor` into two different classes:
1. Input tensor creation (common class for all NLP tasks).
2. Expected Output creation (each NLP task has its own implementation).

This way, we could reuse the class that creates the Input tensor in the plugins.

> **Important:** I think that this Processor class was not reused in previous implementations because of an error with Tensor types.

Anyways, I think it would be nice to discuss and define it in a call when possible.

@dpinol @ericmarcos @vanbasten17

Reply:
The 2 TextProcessor classes are almost identical (the only difference is in the call to preprocess which is now only refactored in 1 of the 2 tasks). 
I'm not sure why you mention the 2 Processor classes look more diverse and they could be kept separate (though their maskUnknownTokens & preprocessing call could be factored out)
 

![image](https://user-images.githubusercontent.com/1954955/115970458-be0ed600-a542-11eb-94f6-7021211d44de.png)

Reply:
You are right, the two TextProcessor are almost identical. When I said that the `Processor` classes look more diverse I referred to the `Processor` class of [`ner`](https://github.com/hubtype/botonic/blob/feat/text-processor/packages/botonic-nlp/src/tasks/ner/process/processor.ts) and [`text-classification`](https://github.com/hubtype/botonic/blob/feat/text-processor/packages/botonic-nlp/src/tasks/text-classification/process/processor.ts) tasks in `botonic-nlp-package`.

Reply:
So, what's the plan to unify them? Is there a PR or a ticket?

Reply:
Here is the [ticket](https://linear.app/hubtype/issue/BOT-358/split-preprocessors-into-2-classes). The plan is to split the `Processor` classes into 2 different classes:
1. For creating the input tensor and sequence.
2. For creating the expected output tensor.

This way we can use the input tensor generator class in all NLP tasks and plugins. 
Also, each NLP task will have its own class responsible for generating the expected output tensor.